packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js

```diff
@@ -34,11 +35,30 @@ export default class TrafficController {
       this.services = data;
       return data;
     });
-    if (this.getSubnet()) {
-      this.subnet = this.getSubnet();
-      this.selectedIp = this.subnet;
-      this.model = this.selectedIp;
-      this.getTraffic();
+    this.ip = this.$stateParams.ip;
+    this.dateTime = this.$stateParams.dateTime
+      ? new Date(this.$stateParams.dateTime)
+      : null;
+    if (this.dateTime) {
+      const dateLimit = new Date();
+      dateLimit.setDate(dateLimit.getDate() - 14);
+      if (this.dateTime >= dateLimit) {
+        const customPeriod = {
+          name: 'custom',
+          label: this.$translate.instant(
+            'network_security_dashboard_filter_custom',
+          ),
+        };
+        this.periods.push(customPeriod);
+        this.period = this.periods[this.periods.length - 1];
```

Comment:
Performance of slice is better
```suggestion
        this.period = this.periods.slice(-1);
```